ptvsd/pydevd/_pydevd_bundle/pydevd_frame.py

```diff
@@ -39,6 +39,12 @@ def send_signature_call_trace(*args, **kwargs):
 def handle_breakpoint_condition(py_db, info, breakpoint, new_frame):
     condition = breakpoint.condition
     try:
+        if breakpoint.handle_hit_condition():
+            return True
+
+        if condition is None:
+            return False
+
         return eval(condition, new_frame.f_globals, new_frame.f_locals)
```

Comment:
Or maybe we should just pass `new_frame` to `handle_hit_condition`, and then do the eval there in the same way it happens here? This would potentially allow for hit conditions that are dynamic (e.g. depend on the value of a local variable), which is a neat feature.

Reply:
This is a good suggestion. I will update it.